### PR TITLE
DAOS-7455 daos: fix destination container creation

### PR DESCRIPTION
**Pull request by dsikich**
When passing a generated cont UUID to filesystem copy
dfs_cont_create was failing. The cont_type was being
set incorrectly when passed to DFS through the container
properties. Also, since uns_resolve_path has been extended
to handle daos://$pool as a destination the cont_clone_hdlr
was updated. The created container UUID is now also being
printed to stdout.

Signed-off-by: Danielle Sikich <danielle.sikich@intel.com>

### CODE REVIEW

**Review by daltonbohning — COMMENTED** [review]
@dsikich I'd recommend adding this pragma to your commit message to run the functional tests as a sanity check:

Features: fs_copy cont_clone

**Comment by dsikich** [comment]
> @dsikich I'd recommend adding this pragma to your commit message to run the functional tests as a sanity check:
> 
> Features: fs_copy cont_clone

@daltonbohning thanks, added it!

File: src/utils/daos_hdlr.c [hunk 1]
```diff
@@ -2751,18 +2761,16 @@ dm_connect(bool is_posix_copy,
...
-
-		/* only set max oid on created dest containers
-		 * if this is a non-posix copy
-		 */
 		if (!is_posix_copy) {
+			props = daos_prop_alloc(2);
+			if (props == NULL) {
+				fprintf(stderr, "Failed to allocate prop (%d)",
+					rc);
+				D_GOTO(out, rc);
```
Comment (daltonbohning) [inline src/utils/daos_hdlr.c:2769]: @dsikich Just noticed... daos_prop_alloc doesn't return an rc
```suggestion
			props = daos_prop_alloc(2);
			if (props == NULL) {
				fprintf(stderr, "Failed to allocate prop\n");
				D_GOTO(out, rc = -DER_NOMEM);
```

**Review by daltonbohning — CHANGES_REQUESTED** [review]

**Review by daltonbohning — APPROVED** [review]
Assuming the tests pass, LGTM.

**Review by mchaarawi — APPROVED** [review]
looks fine to me, although i would really prefer to add some tests for any additional supported features in the daos tool to make sure it work and no regression gets introduced, if for example we change something in the duns.

**Comment by dsikich** [comment]
> looks fine to me, although i would really prefer to add some tests for any additional supported features in the daos tool to make sure it work and no regression gets introduced, if for example we change something in the duns.

@mchaarawi Thanks, there are some functional tests now for fs_copy and cont_clone that were run with Features: fs_copy cont_clone for this PR. There could be more though since there are so many different tools and options for each now. It is getting hard to keep track of and keep them all in sync between the daos tools and mpifileutils.

**Comment by dsikich** [comment]
> > > looks fine to me, although i would really prefer to add some tests for any additional supported features in the daos tool to make sure it work and no regression gets introduced, if for example we change something in the duns.
> > 
> > 
> > @mchaarawi Thanks, there are some functional tests now for fs_copy and cont_clone that were run with Features: fs_copy cont_clone for this PR. There could be more though since there are so many different tools and options for each now. It is getting hard to keep track of and keep them all in sync between the daos tools and mpifileutils.
> 
> well my comment is referring directly to this PR that is fixing this case:
> "When passing a generated cont UUID to filesystem copy dfs_cont_create was failing."
> i don't see a test added for the daos tool, so if this is supposed to be fixing something, it would be good to a test so we don't see a regression.
> 
> same thing for:
> handle daos://$pool as a destination



> > > looks fine to me, although i would really prefer to add some tests for any additional supported features in the daos tool to make sure it work and no regression gets introduced, if for example we change something in the duns.
> > 
> > 
> > @mchaarawi Thanks, there are some functional tests now for fs_copy and cont_clone that were run with Features: fs_copy cont_clone for this PR. There could be more though since there are so many different tools and options for each now. It is getting hard to keep track of and keep them all in sync between the daos tools and mpifileutils.
> 
> well my comment is referring directly to this PR that is fixing this case:
> "When passing a generated cont UUID to filesystem copy dfs_cont_create was failing."
> i don't see a test added for the daos tool, so if this is supposed to be fixing something, it would be good to a test so we don't see a regression.
> 
> same thing for:
> handle daos://$pool as a destination

@mchaarawi Ok, yeah I agree. Would you prefer I add it to this PR?

**Comment by mchaarawi** [comment]
> @mchaarawi Ok, yeah I agree. Would you prefer I add it to this PR?

oh no, that's fine. we can do it in a future PR. i was just saying since im making some changes to the UNS as you can see and it would be good to know that i don't break stuff by running unit tests in CI.

**Comment by dsikich** [comment]
> > @mchaarawi Ok, yeah I agree. Would you prefer I add it to this PR?
> 
> oh no, that's fine. we can do it in a future PR. i was just saying since im making some changes to the UNS as you can see and it would be good to know that i don't break stuff by running unit tests in CI.

@mchaarawi ok, just wanted to make sure. I am going to look into it this week.

**Pull request merged**